src/wallet/ethUtils.ts

```diff
@@ -0,0 +1,166 @@
+import {
+	type Address,
+	type Chain,
+	createPublicClient,
+	formatEther,
+	formatUnits,
+	type Hex,
+	isAddress,
+	type LocalAccount,
+	parseEther,
+	parseUnits,
+	type TransactionRequest,
+} from 'viem'
+import type { ChainTransport } from '../types.js'
+
+export interface SendEthOptions {
+	/** Amount to send in ETH (e.g., "0.1" for 0.1 ETH) */
+	amount: string
+	/** Destination address */
+	to: Address
+	/** Optional gas limit (defaults to estimated) */
+	gas?: bigint
+	/** Optional gas price in gwei (defaults to network rate) */
+	gasPrice?: string
+	/** Optional nonce (defaults to next available) */
+	nonce?: number
+	/** Optional transaction data */
+	data?: Hex
+}
+
+export interface GetBalanceOptions {
+	/** Address to get balance for (defaults to first account) */
+	address?: Address
+	/** Return format - 'eth' for ETH units, 'wei' for Wei units */
+	unit?: 'eth' | 'wei'
+}
+
+/**
+ * Validates an Ethereum address
+ */
+export function validateAddress(address: string): Address {
+	if (!isAddress(address)) {
+		throw new Error(`Invalid Ethereum address: ${address}`)
+	}
+	return address
+}
+
+/**
+ * Converts ETH amount to Wei
+ */
+export function ethToWei(eth: string): bigint {
+	try {
+		// Check for obvious invalid values first
+		if (!eth || eth.trim() === '' || !/^-?\d*\.?\d+$/.test(eth.trim())) {
+			throw new Error('Invalid format')
+		}
+		return parseEther(eth)
+	} catch (_error) {
+		throw new Error(`Invalid ETH amount: ${eth}`)
+	}
+}
+
+/**
+ * Converts Wei amount to ETH
+ */
+export function weiToEth(wei: bigint): string {
+	return formatEther(wei)
+}
+
+/**
+ * Converts Gwei amount to Wei
+ */
+export function gweiToWei(gwei: string): bigint {
+	try {
+		// Check for obvious invalid values first
+		if (!gwei || gwei.trim() === '' || !/^-?\d*\.?\d+$/.test(gwei.trim())) {
+			throw new Error('Invalid format')
+		}
+		return parseUnits(gwei, 9)
+	} catch (_error) {
+		throw new Error(`Invalid Gwei amount: ${gwei}`)
+	}
+}
+
+/**
+ * Converts Wei amount to Gwei
+ */
+export function weiToGwei(wei: bigint): string {
+	return formatUnits(wei, 9)
+}
+
+/**
+ * Gets the balance of an address
+ */
+export async function getAddressBalance(
+	address: Address,
+	getChainTransport: () => ChainTransport,
+	getChain: () => Chain,
+	unit: 'eth' | 'wei' = 'eth',
+): Promise<string> {
+	const publicClient = createPublicClient({
+		chain: getChain(),
+		transport: getChainTransport,
+	})
+
+	const balance = await publicClient.getBalance({ address })
+
+	return unit === 'eth' ? weiToEth(balance) : balance.toString()
+}
```

Comment:
statically initialize the client, not on every call

I'm also not sure why this should be the part of the library?